### PR TITLE
feat: use .Net8.0

### PR DESCRIPTION
**Pull request by aneojgurhem**

### CODE REVIEW

**Review by ngruelaneo — APPROVED** [review]

**Comment by aneojgurhem** [comment]
As it is now, .Net8 is not supported with QEMU. See:
- https://github.com/dotnet/runtime/issues/78340
- https://github.com/dotnet/core/blob/main/release-notes/8.0/supported-os.md#qemu
- https://github.com/NuGet/Home/issues/12227

Therefore, we won't be able to build ARM images with our current workflows.

**Comment by aneojgurhem** [comment]
To build dotnet 8 ARM images, cross compilation should be used such as : https://devblogs.microsoft.com/dotnet/improving-multiplatform-container-support/

**Comment by aneojgurhem** [comment]
Perf tests see https://github.com/aneoconsulting/ArmoniK.Core/pull/484#issuecomment-1713220167 for details.

htc mock (throughput) 620 with dotnet 8 vs 558 tasks/s on main (3b393d17)

File: Tests/Bench/Server/src/Dockerfile [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
-FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
+FROM --platform=$BUILDPLATFORM mcr.microsoft.com/dotnet/sdk:8.0 AS build
+ARG TARGETARCH
+ARG VERSION=1.0.0.0
+
 WORKDIR /src
 COPY ["Tests/Bench/Server/src/ArmoniK.Samples.Bench.Server.csproj", "Tests/Bench/Server/src/"]
 COPY ["Common/src/ArmoniK.Core.Common.csproj", "Common/src/"]
-RUN dotnet restore "Tests/Bench/Server/src/ArmoniK.Samples.Bench.Server.csproj"
+RUN dotnet restore -a ${TARGETARCH} "Tests/Bench/Server/src/ArmoniK.Samples.Bench.Server.csproj"
```
Comment (lemaitre-aneo) [inline Tests/Bench/Server/src/Dockerfile:13]:
```suggestion
RUN dotnet restore -a "${TARGETARCH}" "Tests/Bench/Server/src/ArmoniK.Samples.Bench.Server.csproj"
```

File: Tests/Bench/Server/src/Dockerfile [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
 COPY ["Tests/Bench/Server/src", "Tests/Bench/Server/src"]
 COPY ["Common/src", "Common/src"]
 WORKDIR "/src/Tests/Bench/Server/src"
-RUN dotnet build "ArmoniK.Samples.Bench.Server.csproj" -c Release -o /app/build -p:RunAnalyzers=false -p:WarningLevel=0
-
-FROM build AS publish
-ARG VERSION=1.0.0.0
-RUN dotnet publish "ArmoniK.Samples.Bench.Server.csproj" -c Release -o /app/publish -p:PackageVersion=$VERSION -p:Version=$VERSION
+RUN dotnet publish -a ${TARGETARCH} "ArmoniK.Samples.Bench.Server.csproj" -o /app/publish -p:RunAnalyzers=false -p:WarningLevel=0 -p:PackageVersion=$VERSION -p:Version=$VERSION
```
Comment (lemaitre-aneo) [inline Tests/Bench/Server/src/Dockerfile:17]:
```suggestion
RUN dotnet publish -a "${TARGETARCH}" "ArmoniK.Samples.Bench.Server.csproj" -o /app/publish -p:RunAnalyzers=false -p:WarningLevel=0 -p:PackageVersion="$VERSION" -p:Version="$VERSION"
```

File: Tests/HtcMock/Client/src/Dockerfile [hunk 3]
```diff
@@ -1,31 +1,30 @@
...
+ARG VERSION=1.0.0.0
+
 WORKDIR /src
 COPY ["Tests/HtcMock/Client/src/ArmoniK.Samples.HtcMock.Client.csproj", "Tests/HtcMock/Client/src/"]
 COPY ["Tests/Common/Client/src/ArmoniK.Core.Common.Tests.Client.csproj", "Tests/Common/Client/src/"]
 COPY ["Utils/src/ArmoniK.Core.Utils.csproj", "Utils/src/"]
 COPY ["Base/src/ArmoniK.Core.Base.csproj", "Base/src/"]
 COPY ["Common/src/ArmoniK.Core.Common.csproj", "Common/src/"]
-RUN dotnet restore "Tests/HtcMock/Client/src/ArmoniK.Samples.HtcMock.Client.csproj"
+RUN dotnet restore -a ${TARGETARCH} "Tests/HtcMock/Client/src/ArmoniK.Samples.HtcMock.Client.csproj"
```
Comment (lemaitre-aneo) [inline Tests/HtcMock/Client/src/Dockerfile:16]:
```suggestion
RUN dotnet restore -a "${TARGETARCH}" "Tests/HtcMock/Client/src/ArmoniK.Samples.HtcMock.Client.csproj"
```

File: Tests/HtcMock/Client/src/Dockerfile [hunk 4]
```diff
@@ -1,31 +1,30 @@
...
 COPY ["Utils/src/", "Utils/src/"]
 COPY ["Base/src/", "Base/src/"]
 COPY ["Common/src", "Common/src"]
 WORKDIR "/src/Tests/HtcMock/Client/src"
-RUN dotnet build "ArmoniK.Samples.HtcMock.Client.csproj" -c Release -o /app/build -p:RunAnalyzers=false -p:WarningLevel=0
-
-FROM build AS publish
-ARG VERSION=1.0.0.0
-RUN dotnet publish "ArmoniK.Samples.HtcMock.Client.csproj" -c Release -o /app/publish -p:PackageVersion=$VERSION -p:Version=$VERSION
+RUN dotnet publish -a ${TARGETARCH} "ArmoniK.Samples.HtcMock.Client.csproj" --no-restore -o /app/publish -p:RunAnalyzers=false -p:WarningLevel=0 -p:PackageVersion=$VERSION -p:Version=$VERSION
```
Comment (lemaitre-aneo) [inline Tests/HtcMock/Client/src/Dockerfile:23]:
```suggestion
RUN dotnet publish -a "${TARGETARCH}" "ArmoniK.Samples.HtcMock.Client.csproj" --no-restore -o /app/publish -p:RunAnalyzers=false -p:WarningLevel=0 -p:PackageVersion="$VERSION" -p:Version="$VERSION"
```

File: Tests/HtcMock/Server/src/Dockerfile [hunk 5]
```diff
@@ -1,25 +1,24 @@
...
-FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
+FROM --platform=$BUILDPLATFORM mcr.microsoft.com/dotnet/sdk:8.0 AS build
+ARG TARGETARCH
+ARG VERSION=1.0.0.0
+
 WORKDIR /src
 COPY ["Tests/HtcMock/Server/src/ArmoniK.Samples.HtcMock.Server.csproj", "Tests/HtcMock/Server/src/"]
 COPY ["Common/src/ArmoniK.Core.Common.csproj", "Common/src/"]
-RUN dotnet restore "Tests/HtcMock/Server/src/ArmoniK.Samples.HtcMock.Server.csproj"
+RUN dotnet restore -a ${TARGETARCH} "Tests/HtcMock/Server/src/ArmoniK.Samples.HtcMock.Server.csproj"
```
Comment (lemaitre-aneo) [inline Tests/HtcMock/Server/src/Dockerfile:13]:
```suggestion
RUN dotnet restore -a "${TARGETARCH}" "Tests/HtcMock/Server/src/ArmoniK.Samples.HtcMock.Server.csproj"
```

File: Tests/HtcMock/Server/src/Dockerfile [hunk 6]
```diff
@@ -1,25 +1,24 @@
...
 COPY ["Tests/HtcMock/Server/src", "Tests/HtcMock/Server/src"]
 COPY ["Common/src", "Common/src"]
 WORKDIR "/src/Tests/HtcMock/Server/src"
-RUN dotnet build "ArmoniK.Samples.HtcMock.Server.csproj" -c Release -o /app/build -p:RunAnalyzers=false -p:WarningLevel=0
-
-FROM build AS publish
-ARG VERSION=1.0.0.0
-RUN dotnet publish "ArmoniK.Samples.HtcMock.Server.csproj" -c Release -o /app/publish -p:PackageVersion=$VERSION -p:Version=$VERSION
+RUN dotnet publish -a ${TARGETARCH} "ArmoniK.Samples.HtcMock.Server.csproj" -o /app/publish -p:RunAnalyzers=false -p:WarningLevel=0 -p:PackageVersion=$VERSION -p:Version=$VERSION
```
Comment (lemaitre-aneo) [inline Tests/HtcMock/Server/src/Dockerfile:17]:
```suggestion
RUN dotnet publish -a "${TARGETARCH}" "ArmoniK.Samples.HtcMock.Server.csproj" -o /app/publish -p:RunAnalyzers=false -p:WarningLevel=0 -p:PackageVersion="$VERSION" -p:Version="$VERSION"
```

File: Tests/Stream/Client/Dockerfile [hunk 7]
```diff
@@ -1,22 +1,26 @@
...
 
-FROM base AS build
 WORKDIR /src
 COPY ["Tests/Stream/Client/ArmoniK.Extensions.Common.StreamWrapper.Tests.Client.csproj", "Tests/Stream/Client/"]
 COPY ["Tests/Stream/Common/ArmoniK.Extensions.Common.StreamWrapper.Tests.Common.csproj", "Tests/Stream/Common/"]
 COPY ["Utils/src/ArmoniK.Core.Utils.csproj", "Utils/src/"]
 COPY ["Base/src/ArmoniK.Core.Base.csproj", "Base/src/"]
 COPY ["Common/src/ArmoniK.Core.Common.csproj", "Common/src/"]
-RUN dotnet restore "Tests/Stream/Client/ArmoniK.Extensions.Common.StreamWrapper.Tests.Client.csproj"
+RUN dotnet restore -a ${TARGETARCH} "Tests/Stream/Client/ArmoniK.Extensions.Common.StreamWrapper.Tests.Client.csproj"
```
Comment (lemaitre-aneo) [inline Tests/Stream/Client/Dockerfile:16]:
```suggestion
RUN dotnet restore -a "${TARGETARCH}" "Tests/Stream/Client/ArmoniK.Extensions.Common.StreamWrapper.Tests.Client.csproj"
```

File: Tests/Stream/Client/Dockerfile [hunk 8]
```diff
@@ -1,22 +1,26 @@
...
 COPY ["Tests/Stream/Client", "Tests/Stream/Client"]
 COPY ["Tests/Stream/Common", "Tests/Stream/Common"]
 COPY ["Utils/src/", "Utils/src/"]
 COPY ["Base/src/", "Base/src/"]
 COPY ["Common/src", "Common/src"]
 WORKDIR "/src/Tests/Stream/Client"
-RUN dotnet build "ArmoniK.Extensions.Common.StreamWrapper.Tests.Client.csproj" -c Release -o /app/build -p:RunAnalyzers=false -p:WarningLevel=0
+RUN dotnet build "ArmoniK.Extensions.Common.StreamWrapper.Tests.Client.csproj" -a ${TARGETARCH} -c Release -o /app/build -p:RunAnalyzers=false -p:WarningLevel=0
```
Comment (lemaitre-aneo) [inline Tests/Stream/Client/Dockerfile:23]:
```suggestion
RUN dotnet build "ArmoniK.Extensions.Common.StreamWrapper.Tests.Client.csproj" -a "${TARGETARCH}" -c Release -o /app/build -p:RunAnalyzers=false -p:WarningLevel=0
```

File: Tests/Stream/Server/Dockerfile [hunk 9]
```diff
@@ -1,30 +1,29 @@
...
+FROM --platform=$BUILDPLATFORM mcr.microsoft.com/dotnet/sdk:8.0 AS build
+ARG TARGETARCH
+ARG VERSION=1.0.0.0
+
 WORKDIR /src
 COPY ["Tests/Stream/Server/ArmoniK.Extensions.Common.StreamWrapper.Tests.Server.csproj", "Tests/Stream/Server/"]
 COPY ["Tests/Stream/Common/ArmoniK.Extensions.Common.StreamWrapper.Tests.Common.csproj", "Tests/Stream/Common/"]
 COPY ["Common/src/ArmoniK.Core.Common.csproj", "Common/src/"]
-RUN dotnet restore "Tests/Stream/Server/ArmoniK.Extensions.Common.StreamWrapper.Tests.Server.csproj"
+RUN dotnet restore -a ${TARGETARCH} "Tests/Stream/Server/ArmoniK.Extensions.Common.StreamWrapper.Tests.Server.csproj"
```
Comment (lemaitre-aneo) [inline Tests/Stream/Server/Dockerfile:17]:
```suggestion
RUN dotnet restore -a "${TARGETARCH}" "Tests/Stream/Server/ArmoniK.Extensions.Common.StreamWrapper.Tests.Server.csproj"
```

File: Tests/Stream/Server/Dockerfile [hunk 10]
```diff
@@ -1,30 +1,29 @@
...
 COPY ["Tests/Stream/Server", "Tests/Stream/Server"]
 COPY ["Tests/Stream/Common", "Tests/Stream/Common"]
 COPY ["Common/src", "Common/src"]
 WORKDIR "/src/Tests/Stream/Server"
-RUN dotnet build "ArmoniK.Extensions.Common.StreamWrapper.Tests.Server.csproj" -c Release -o /app/build -p:RunAnalyzers=false -p:WarningLevel=0
-
-FROM build AS publish
-ARG VERSION=1.0.0.0
-RUN dotnet publish "ArmoniK.Extensions.Common.StreamWrapper.Tests.Server.csproj" -c Release -o /app/publish -p:PackageVersion=$VERSION -p:Version=$VERSION
+RUN dotnet publish -a ${TARGETARCH} "ArmoniK.Extensions.Common.StreamWrapper.Tests.Server.csproj" --no-restore -o /app/publish -p:RunAnalyzers=false -p:WarningLevel=0 -p:PackageVersion=$VERSION -p:Version=$VERSION
```
Comment (lemaitre-aneo) [inline Tests/Stream/Server/Dockerfile:22]:
```suggestion
RUN dotnet publish -a "${TARGETARCH}" "ArmoniK.Extensions.Common.StreamWrapper.Tests.Server.csproj" --no-restore -o /app/publish -p:RunAnalyzers=false -p:WarningLevel=0 -p:PackageVersion="$VERSION" -p:Version="$VERSION"
```

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**